warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ValidPatternVisitor.java

```diff
@@ -69,22 +59,65 @@ public Object visit(ASTERNode node, Object data) {
      */
     @Override
     public Object visit(ASTNRNode node, Object data) {
-        Object literalValue;
+        parseAndPutPattern(node);
+        return data;
+    }
+    
+    /**
+     * Visit an ASTFunctionNode to catch cases like #INCLUDE or #EXCLUDE that accept a regex as an argument
+     * 
+     * @param node
+     * @param data
+     * @return
+     */
+    @Override
+    public Object visit(ASTFunctionNode node, Object data) {
         
-        // Catch the situation where a user might enter FIELD1 !~ VALUE1
-        try {
-            literalValue = JexlASTHelper.getLiteralValue(node);
-        } catch (NoSuchElementException e) {
-            return data;
+        // Should pull back an EvaluationPhaseFilterFunctionsDescriptor
+        JexlArgumentDescriptor descriptor = JexlFunctionArgumentDescriptorFactory.F.getArgumentDescriptor(node);
+        if (descriptor == null) {
+            throw new IllegalStateException("Could not get descriptor for ASTFunctionNode");
         }
         
+        if (descriptor.regexArguments()) {
+            // Extract the args for this function
+            FunctionJexlNodeVisitor functionVisitor = new FunctionJexlNodeVisitor();
+            functionVisitor.visit(node, null);
+            List<JexlNode> args = functionVisitor.args();
+            for (JexlNode arg : args) {
+                // Only take the literals
+                if (arg instanceof ASTStringLiteral) {
+                    parseAndPutPattern(arg);
+                }
+            }
+        }
+        // Do not descend to children, the ValidPatternVisitor views a function node as a leaf node.
+        return data;
+    }
+    
+    /**
+     * Parse a literal value and put into the pattern cache if it does not exist.
+     *
+     * @param node
+     */
+    public void parseAndPutPattern(JexlNode node) {
+        // Catch the situation where a user might enter FIELD1 !~ VALUE1
+        Object literalValue = JexlASTHelper.getLiteralValue(node);
         if (literalValue != null && String.class.equals(literalValue.getClass())) {
             String literalString = (String) literalValue;
-            if (patternCache.containsKey(literalString)) {
-                return data;
+            // Only parse literals that contain a regex char
+            if (StringUtils.containsAny(literalString, "?.*^+-_[](){}")) {
```

Comment:
Tried to scope down the number of patterns that get parsed. Might not be necessary.

Reply:
I don't think this is necessary, and in fact could be problematic.  A Regex is not guaranteed to contain one of those characters.  Perhaps a better way to go is if the literal contains only alphanumeric characters.